compvits/scripts/train_linear_head.py

```diff
@@ -0,0 +1,251 @@
+from torch.utils.data import Dataset
+import numpy as np
+import re
+import os
+import argparse
+import json
+from pathlib import Path
+
+import torch
+from torch import nn
+import torch.distributed as dist
+import torch.backends.cudnn as cudnn
+from torchvision import datasets
+
+from compvits import utils
+from compvits.utils import MetricLogger
+
+
+class PickleDataset(Dataset):
+    def __init__(self, root_dir):
+        features_pattern = r"rank\d+_chunk\d+_train_lastCLS_features.npy"
+        targets_pattern = r"rank\d+_chunk\d+_train_lastCLS_targets.npy"
+
+        features_files = []
+        target_files = []
+        for filename in os.listdir(root_dir):
+            # Check if the filename matches the pattern
+            if re.match(features_pattern, filename):
+                features_files.append(filename)
+            if re.match(targets_pattern, filename):
+                target_files.append(filename)
+
+        features = [torch.tensor(np.load(os.path.join(root_dir, f),
+                                         allow_pickle=True)) for f in features_files]
+        targets = [torch.tensor(np.load(os.path.join(root_dir, f),
+                                        allow_pickle=True)) for f in target_files]
+
+        self.features = torch.cat(features, dim=0)
+        self.targets = torch.cat(targets, dim=0)
+
+    def __len__(self):
+        return self.features.shape[0]
+
+    def __getitem__(self, idx):
+        return self.features[idx].float(), self.targets[idx]
+
+
+class LinearClassifier(nn.Module):
+    """Linear layer to train on top of frozen features"""
+    def __init__(self, dim, num_labels=1000):
+        super(LinearClassifier, self).__init__()
+        self.num_labels = num_labels
+        self.linear = nn.Linear(dim, num_labels)
+        self.linear.weight.data.normal_(mean=0.0, std=0.01)
+        self.linear.bias.data.zero_()
+
+    def forward(self, x):
+        # flatten
+        x = x.view(x.size(0), -1)
```

Comment:
Unnecessary? We loose structure of the input, not a big deal probably, but still.
Also it doesn't provide computational benefit if data was already contiguous (and this code fails it data is not contiguous anyway)

x = torch.arange(0,9).reshape(3,3)
y = torch.permute(x, dims=(1,0))
print(x.is_contiguous()) # True
print(y.is_contiguous()) # False
y.view(9,1) # Runtime error

Reply:
Also here we follow DINO codebase